### PR TITLE
Add template-nuget file DynamoSamples

### PR DESCRIPTION
**Pull request by ZiyunShang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Add template-nuget file of DynamoSamples so that it can create Dynamo Sample files nuget package and push to nuget.org in CICD.

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang @AndyDu1985 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoSamples.nuspec [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
+    <id>DynamoVisualProgramming.DynamoSamples</id>
+    <version>@VERSION@</version>
+    <authors>Autodesk</authors>
+    <owners>Autodesk</owners>
+    <licenseUrl>http://opensource.org/licenses/MIT</licenseUrl>
```
Comment (mjkkirschner) [inline tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoSamples.nuspec]: I think most dynamo assets are licensed with apache 2.0 license.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoSamples.nuspec [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    <projectUrl>https://github.com/DynamoDS/Dynamo</projectUrl>
+    <iconUrl>https://raw.githubusercontent.com/DynamoDS/Dynamo/master/doc/distrib/Images/logo_square_32x32.png</iconUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>This package bundles all Sample files from the Dynamo doc distrib Samples folder.</description>
+    <copyright>Copyright Autodesk 2016</copyright>
```
Comment (mjkkirschner) [inline tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoSamples.nuspec]: 2018?

**Comment by QilongTang** [comment]
@ZiyunShang @AndyDu1985 Not sure if you noticed, gallery in Dynamo also need localization. It is usually installed side by side with Samples.

File: tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoSamples.nuspec (hunk 1 above)
Comment (ZiyunShang) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoSamples.nuspec]: On other templates, LicenseUrl is still MIT. Which one should I choose?

Comment (mjkkirschner) [reply tools/NuGet/template-nuget/DynamoVisualProgramming.DynamoSamples.nuspec]: hmm, lets keep it consistent then and we'll decide in the future if they should all be changed over at once. Thanks for checking it out!

**Comment by ZiyunShang** [comment]
Hi @QilongTang , I will add the gallery to this template, and put it together with Samples.

**Comment by ZiyunShang** [comment]
Hi @QilongTang @mjkkirschner , is there anything else I need to do?

**Comment by QilongTang** [comment]
@ZiyunShang Looks good to me! :shipit:

**Comment by mjkkirschner** [comment]
@ZiyunShang thanks! LGTM.

**Comment by AndyDu1985** [comment]
@mjkkirschner @QilongTang Is it OK merge this PR? Or there is anything additional need to be done?

**Comment by ZiyunShang** [comment]
Hi @QilongTang @mjkkirschner , could you help me merge this?

**Pull request merged**